developer_guides/fuzzing/testbench_afl_fuzzing.rst

```diff
@@ -0,0 +1,88 @@
+.. _testbench-afl-fuzzing:
+
+Build a Fuzzing Testbench with AFL
+##################################
+
+American fuzzy lop (AFL) is a free software fuzzer that can be used to detect software bugs. Use these instructions to build and run a testbench with AFL.
+
+Install AFL
+***********
+
+Follow the steps in the `AFL Quick Start Guide <https://github.com/google/AFL/blob/master/docs/QuickStartGuide.txt>`_ to install AFL on your system.
+
+We assume that AFL is installed at:
+
+::
+
+   $HOME/work/
+
+
+Build a testbench with AFL instrumentation
+******************************************
+
+According to AFL's `README <https://github.com/google/AFL/blob/master/README.md>`_, AFL is a "brute-force fuzzer coupled with an exceedingly
+simple but rock-solid instrumentation-guided genetic algorithm." Add
+instrumentation to the code before running a fuzzer on it in order to get
```

Comment:
I feel here we need to specify it's really important to add instrumentation.

When i ran fuzzer on some parts of code without instrumentation i didn't get any results. But on adding the instrumentation, there were some results. So, i feel it's a bit necessary to say the reader that it's important to instrument before running the fuzzer.

Reply:
Updated to:
You must add instrumentation to the code before running a fuzzer in order to get potentially useful results; otherwise, you might not get any results.

*Note that I changed it to "potentially useful" results as opposed to "good" because we really cannot guarantee that the results they get will be good.